build.sh

```diff
@@ -221,20 +225,7 @@ esac
 __BuildOS=$__HostOS
 __BuildType=Debug
 __CMakeArgs=DEBUG
-
-case $__HostOS in
-    FreeBSD)
-        __monoroot=/usr/local
-        ;;
-    OSX)
-        __monoroot=/Library/Frameworks/Mono.framework/Versions/Current
-        ;;
-    *)
-        __monoroot=/usr
-        ;;
-esac
-
-__referenceassemblyroot=$__monoroot/lib/mono/xbuild-frameworks
+__referenceassemblyroot="$(csharp FindMonoXBuildFrameworks.cs)"
```

Comment:
"Hardcoding" the location of the system-level Mono was actually intentional. The reason is that only the Mono from Xamarin packages ship with the PCL reference assemblies, so e.g. if I have a custom built Mono in my PATH I still want the corefx build to use those PCL assemblies from the regular Mono install. This change would break that workflow.

I'd prefer having a way to override `__referenceassemblyroot` instead.